docs/source/performance.md

```diff
@@ -212,7 +213,26 @@ Then your software could have special memory needs. For example, when generating
 
 For convolutions and linear layers there are 2x flops in the backward compared to the forward, which generally translates into ~2x slower (sometimes more, because sizes in the backward tend to be more awkward). Activations are usually bandwidth-limited, and it’s typical for an activation to have to read more data in the backward than in the forward (e.g. activation forward reads once, writes once, activation backward reads twice, gradOutput and output of the forward, and writes once, gradInput).
 
-### fp16
+
+### Floating Data Types
+
+Here are the commonly used floating point data types choice of which impacts both memory usage and throughput:
+
+- fp32 (`float32`)
+- fp16 (`float16`)
+- bf16 (`bfloat16`)
+- tf32 (CUDA internal data type)
+
+Here is a diagram that shows how these data types correlate to each other.
+
+![data types](imgs/tf32-bf16-fp16-fp32.png)
```

Comment:
```suggestion
![data types](/transformers/_images/tf32-bf16-fp16-fp32.png)
```
Image links need to look like this now.

Reply:
why would you change this now just for this image? 

I'm asking since all the other imgs are still at the old style:

```
docs/source/perplexity.rst:.. image:: imgs/ppl_full.gif
docs/source/perplexity.rst:.. image:: imgs/ppl_chunked.gif
docs/source/perplexity.rst:.. image:: imgs/ppl_sliding.gif
docs/source/parallelism.md:![DeepSpeed-Image-1](imgs/parallelism-zero.png)
docs/source/parallelism.md:![mp-pp](imgs/parallelism-gpipe-bubble.png)
docs/source/parallelism.md:![interleaved-pipeline-execution](imgs/parallelism-sagemaker-interleaved-pipeline.png)
docs/source/parallelism.md:![Parallel GEMM](imgs/parallelism-tp-parallel_gemm.png)
docs/source/parallelism.md:![independent GeLU](imgs/parallelism-tp-independent-gelu.png)
docs/source/parallelism.md:![parallel shard processing](imgs/parallelism-tp-parallel_shard_processing.png)
docs/source/parallelism.md:![parallel self-attention](imgs/parallelism-tp-parallel_self_attention.png)
docs/source/parallelism.md:![dp-pp-2d](imgs/parallelism-zero-dp-pp.png)
docs/source/parallelism.md:![dp-pp-tp-3d](imgs/parallelism-deepspeed-3d.png)
docs/source/parallelism.md:![flex-flow-soap](imgs/parallelism-flexflow.jpeg)
docs/source/main_classes/optimizer_schedules.rst:.. image:: /imgs/warmup_constant_schedule.png
docs/source/main_classes/optimizer_schedules.rst:    :target: /imgs/warmup_constant_schedule.png
docs/source/main_classes/optimizer_schedules.rst:.. image:: /imgs/warmup_cosine_schedule.png
docs/source/main_classes/optimizer_schedules.rst:    :target: /imgs/warmup_cosine_schedule.png
docs/source/main_classes/optimizer_schedules.rst:.. image:: /imgs/warmup_cosine_hard_restarts_schedule.png
docs/source/main_classes/optimizer_schedules.rst:    :target: /imgs/warmup_cosine_hard_restarts_schedule.png
docs/source/main_classes/optimizer_schedules.rst:.. image:: /imgs/warmup_linear_schedule.png
docs/source/main_classes/optimizer_schedules.rst:    :target: /imgs/warmup_linear_schedule.png
docs/source/model_summary.rst:.. image:: imgs/local_attention_mask.png
docs/source/add_new_model.rst:.. image:: ./imgs/transformers_overview.png
```

Reply:
You can ignore the rst files as the path there is corrected when the `doc-builder` converts them to MarkDown (as we progressively convert everything to MarkDown they will disappear).

For the `parallelism.md` file I think you didn't have the latest master :-)

Reply:
oh, it just changed! got it - rebased it - all is good - thank you!